src/lib/api/api.go

```diff
@@ -196,25 +198,54 @@ func (a *API) Query(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if hash == "" {
+		switch Type {
+		case "system":
+			timeSeriesPath = "storage/system" + tsdb.FileExtension
+		case "journal":
+			timeSeriesPath = "storage/journal" + tsdb.FileExtension
+		default:
+			a.send(w, []byte("INVALID_TYPE"))
+			return
+		}
+		if ok := tsdb.VerifyChainPathExists(timeSeriesPath); !ok {
+			a.send(w, []byte("INVALID_PATH"))
+			return
+		}
+		newBlocks = []tsdb.Block{}
+	} else {
+		if _, ok := (*a.Matrix)[hash]; !ok {
+			a.send(w, []byte("INVALID_PATH"))
+			return
+		}
+		matrix := (*a.Matrix)[hash]
+
+		switch Type {
+		case "ping":
+			newBlocks = matrix.PingChain.Chain
+		case "jitter":
+			newBlocks = matrix.JitterChain.Chain
+		default:
```

Comment:
Will `Type` not have `monitoring`?

Reply:
Because `Query` is used for Testing which only has Ping and Jitter for now, so I just made cases for them. I will include monitoring also for future purposes.